pepper-apis/src/main/java/org/broadinstitute/ddp/filter/RateLimitFilter.java

```diff
@@ -0,0 +1,40 @@
+package org.broadinstitute.ddp.filter;
+
+import java.util.Map;
+import javax.servlet.ServletException;
+
+import org.broadinstitute.ddp.exception.DDPException;
+import org.eclipse.jetty.servlets.DoSFilter;
+import spark.Filter;
+import spark.Request;
+import spark.Response;
+
+public class RateLimitFilter implements Filter {
+
+    private final int maxRequestsPerSecond;
+
+    private final int burst;
+
+    private final DoSFilter rateLimitFilter;
+
+    public RateLimitFilter(int maxRequestsPerSecond, int burst) {
+        this.maxRequestsPerSecond = maxRequestsPerSecond;
+        this.burst = burst;
+        DoSFilter rateLimitFilter = new DoSFilter();
+        rateLimitFilter.setEnabled(true);
+        try {
+            rateLimitFilter.init(new FilterConfig("DOSFilter", Map.of("delayMs", "-1",
```

Comment:
We should set `delayMs` to `0` (zero) to match the `nodelay` option in nginx for better experience.